test/e2e/scenario/recorder/viewports.scenario.ts

```diff
@@ -0,0 +1,351 @@
+        const navBarHeightChange = nextVisualViewport.height - initialVisualViewport.height
+        expect(navBarHeightChange).toBeLessThanOrEqual(NAVBAR_HEIGHT_CHANGE_UPPER_BOUND)
+
+        // Visual Viewport Scroll should change without visual viewport affect
+        expectToBeNearby(lastScrollData.x, initialScrollX + LAYOUT_SCROLL_AMOUNT)
+        expectToBeNearby(lastScrollData.y, initialScrollY + LAYOUT_SCROLL_AMOUNT)
+        expectToBeNearby(lastScrollData.x, nextScrollX)
+        expectToBeNearby(lastScrollData.y, nextScrollY)
+      })
+  })
+
+  describe('visual viewport properties', () => {
+    createTest('pinch zoom "scroll" event reports visual viewport position')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        const VISUAL_SCROLL_DOWN_PX = 100
+        await buildScrollablePage()
+        await performSignificantZoom()
+        await visualScrollVerticallyDown(VISUAL_SCROLL_DOWN_PX)
+        const nextVisualViewportDimension = await getVisualViewport()
+        const lastVisualViewportRecord = await getLastRecord(events, findAllVisualViewports)
+        expectToBeNearby(lastVisualViewportRecord.data.pageTop, nextVisualViewportDimension.pageTop)
+      })
+
+    createTest('pinch zoom "resize" event reports visual viewport scale')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        await performSignificantZoom()
+        const nextVisualViewportDimension = await getVisualViewport()
+        const lastVisualViewportRecord = await getLastRecord(events, findAllVisualViewports)
+        expectToBeNearby(lastVisualViewportRecord.data.scale, nextVisualViewportDimension.scale)
+      })
+  })
+})
+
+function getLastSegment(events: EventRegistry) {
+  return events.sessionReplay[events.sessionReplay.length - 1].segment.data
+}
+
+function initRumAndStartRecording(initConfiguration: RumInitConfiguration) {
+  window.DD_RUM!.init(initConfiguration)
+  window.DD_RUM!.startSessionReplayRecording()
+}
+
+const isGestureUnsupported = () => {
+  const { capabilities } = browser
+  return (
+    capabilities.browserName === 'firefox' ||
+    capabilities.browserName === 'Safari' ||
+    capabilities.browserName === 'msedge' ||
+    capabilities.platformName === 'windows' ||
+    capabilities.platformName === 'linux'
+  )
+}
+
+// Flakiness: Working with viewport sizes has variations per device of a few pixels
+function expectToBeNearby(numA: number, numB: number) {
+  const roundedA = Math.round(numA)
+  const roundedB = Math.round(numB)
+  const test = Math.abs(roundedA - roundedB) <= 5
+  if (test) {
+    expect(test).toBeTruthy()
```

Comment:
This seems useless, maybe remove it?

Reply:
NIT: I would write the whole `expectToBeNearby` assertion as:

```ts
expect(Math.abs(numA - numB)).toBeLessThan(5)
```
(I don't think we need to round numbers)